File: platforms/android/stato/src/main/java/com/facebook/stato/nativeplugins/components/JsonSidebarSection.java
package com.facebook.stato.nativeplugins.components;

import com.facebook.stato.core.StatoObject;

public class JsonSidebarSection implements SidebarSection {
  private final String title;
  private final StatoObject content;

  public JsonSidebarSection(String title, StatoObject content) {
    this.title = title;
    this.content = content;
  }

  @Override
  public StatoObject serialize() {
    return new StatoObject.Builder()
        .put("title", title)
        .put("type", "json")
        .put("content", content)
        .build();
  }
}

File: platforms/android/android-stato/src/main/kotlin/org/stato/nativeplugins/components/JsonSidebarSection.kt
package org.stato.nativeplugins.components

import org.stato.core.StatoObject

class JsonSidebarSection(private val title: String, private val content: StatoObject) : SidebarSection {

    override fun serialize(): StatoObject {
    return StatoObject.Builder()
      .put("title", title)
      .put("type", "json")
      .put("content", content)
      .build()
  }
}
